src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ReferralHelper.java

```diff
@@ -0,0 +1,352 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.helpers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ecm.common.helpers.UtilHelper;
+import uk.gov.hmcts.et.common.model.bulk.types.DynamicFixedListType;
+import uk.gov.hmcts.et.common.model.bulk.types.DynamicValueType;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.items.DateListedTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.items.DocumentTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.items.HearingTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.items.ReferralReplyTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.items.ReferralTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.types.ReferralReplyType;
+import uk.gov.hmcts.et.common.model.ccd.types.ReferralType;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.UserService;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service

```

Comment:
Service or Helper? File should be renamed and moved to appropriate package

Reply:
This is a helper, I am only using @Service annotation so that the class can be auto wired, but can change it to @Component if it makes more sense?